### DISCUSSION TITLE
How to use it?

### DISCUSSION THREAD

**Discussion by Slesa** [wiredwiz/Grun.Net#46] [General]
Hi,
first of all: thanks a lot for this tool. 
But I don't get it how to use it. I have some g4 files, when I load them nothing happens. When I load the assemblies thei are compiled too, I get the message that no grammar was found.

What I urgently need is a tool that tells me, why the if statements do not work, and why some other statements are not recognized. Is this the tool I am looking for?

**Comment by sagasm** [comment]
I wrote two short articles on how to use GrunDotNet and GrunDotNet Standard, but they are in Chinese.

**Comment by Dansko-Thad** [comment]
> Hi, first of all: thanks a lot for this tool. But I don't get it how to use it. I have some g4 files, when I load them nothing happens. When I load the assemblies thei are compiled too, I get the message that no grammar was found.
> 
> What I urgently need is a tool that tells me, why the if statements do not work, and why some other statements are not recognized. Is this the tool I am looking for?

My first thought is that you are probably using the wrong version based on your runtime.  There are 2 different runtimes for ANTLR on C#.  My original version supported the older runtime, whereas the newer versions only supports the newer.  Version 1.3.23340.0 and prior supports the older runtime.  If your assemblies are compiled against the older runtime, but you are using the newest release of the tool, you will get that error message.  If you are using the correct version for the runtime, it should be as simple as just loading the compiled assembly that includes your grammar class, then entering some sample text to parse.  GrunWin will certainly allow you to debug your grammar as desired.  That, in fact, is my main use for the tool.  I'm always happy to help or take suggestions.

I would love to create one version of the software that supports both runtimes seamlessly, but that would require significant redesign so it isn't likely to happen any time soon.
